src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -80,3 +80,48 @@ void sfSoundRecorder_setProcessingInterval(sfSoundRecorder* soundRecorder, sfTim
 {
     CSFML_CALL(soundRecorder, setProcessingInterval(interval));
 }
+
+
+////////////////////////////////////////////////////////////
+const char** sfSoundRecorder_getAvailableDevices(size_t* count)
+{
+    static std::vector<std::string> stringDevices = sf::SoundRecorder::getAvailableDevices();
+    static std::vector<const char*> cstringDevices;
+
+    if (cstringDevices.empty() && !stringDevices.empty())
```

Comment:
This is just a small optimization so we don't continually try to populate the cstringDevices vector when no devices are available.